### PR TITLE
added settings screen

### LINKED ISSUE #74: Add mask selection to the GUI (Library 3)
## User story
1. As a user
2. I want to use the GUI to select the mask in the video. 
3. So that I can choose a mask for Blacklisting or Whitelisting

This ticket should use the functionality provided in ticket  #61.

## Acceptance criteria
* It is possible to upload the mask png when clicking the upload button from the file system
* When the user clicks save and a mask png is selected, the mask should be considered in the video difference computation
* The mask upload functionality should look like this:
* When pressing save the parameters are saved and the advanced settings screen is closed

Image

**Hint: Scope of this ticket is not to implement the hyperparameter slider!**

## Definition of done (DoD)
* Create feature branch
* Implement the functionality
* Opened pull request and assigned reviewer
* If necessary: required changes are implemented
* Delete feature branch
* If necessary: Component Test has been written
* Code review has been completed and code has been merged

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### LINKED ISSUE #108: Accessible hyperparameter settings in the GUI (Library 2, Library 3)
## User story
1. As a user 
2. I want / need to change certain settings/ hyperparameters directly in the GUI
3. So that I can better use the image difference and the result in the GUI.  

## Acceptance criteria
* It is possible to set the two hyperparameters gap open and gap extension via a slider
* The optimal hyperparameter for gap open and gap closed (optimal hyperparameter are the results of ticket #121) should be preselected by default
* It is possible to readjust the default setting for the hyperparameter via the slider
* the hyperparameter that are set when clicking save should be considered in the computation of the difference
* When pressing safe the parameters are saved and the advanced settings screen is closed

Image

**Hint: The upload of the mask is not in the scope of this ticket!**

## Definition of done (DoD)
* Create feature branch
* Implement the functionality
* Opened pull request and assigned reviewer
* If necessary: required changes are implemented
* Delete feature branch
* If necessary: Component Test has been written
* Code review has been completed and code has been merged

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### LINKED ISSUE #143: Add window for advanced settings (Library 3)
## User story
1. As a user 
2. I want / need to be able to open an additional window
3. So that I can change advandces settings

## Acceptance criteria
* On the start page of the GUI, there should be a button to open the advanced settings screen 
* When the user clicks on the advanced settings button a seperate window should open
* The user should be able to close the advanced settings window and go back to the video selection  
* The advaced settings windows is only a skeleton without any buttons and text
* The GUI shoudl look like as shown below
* a safe button is added that closes the screen and goes back to the video selection 

Image

Image

**Hint: Scope of this ticket is not to implement the slider for the hyperparameter and the mask upload functionality**

## Definition of done (DoD)
* Create feature branch
* Implement the functionality
* Opened pull request and assigned reviewer
* If necessary: required changes are implemented
* Delete feature branch
* If necessary: Component Test has been written
* Code review has been completed and code has been merged

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

### PR DESCRIPTION
**Pull request by a-miscellaneous**
image

added button to go to difference screen
added button to save in difference screen
added sliders
added mask selection

closes #143 
closes #108 
closes #74 
TODO: better design

### CODE REVIEW

File: GUI/src/main/kotlin/models/AppState.kt
Comment (akriese) [inline GUI/src/main/kotlin/models/AppState.kt:22]: These defaults are different from what was determined in https://github.com/amosproj/amos2023ws03-gui-frame-diff/pull/137.

File: GUI/src/main/kotlin/ui/screens/SettingsScreen.kt
Comment (akriese) [inline GUI/src/main/kotlin/ui/screens/SettingsScreen.kt:41]: For consistency reasons we should call this gapExtensionPenalty as in lib2.

File: GUI/src/main/kotlin/ui/screens/SettingsScreen.kt
Comment (akriese) [inline GUI/src/main/kotlin/ui/screens/SettingsScreen.kt:51]: two spaces between words

**Review by akriese — CHANGES_REQUESTED** [review]

**Review by akriese — APPROVED** [review]

**Comment by akriese** [comment]
Please wait for decision on adding text input for the parameters before merging :)

**Comment by AlperK61** [comment]
ok, please notify me

**Pull request merged**
